### PR TITLE
feat: enable header to accept custom menus

### PR DESCRIPTION
**Pull request by syedsajjadkazmii**
This PR adds the following functionality

- Accept the main menu in the props. if not provided, the default main menu will be used.
- Accept the secondary menu in the props.
- Accept the user menu in the props.
- Restructure the user menu to add support for menu groups with heading in the user menu.
- Removed `appMenu` from `DesktopHeader` because its usage could not be found. ([reference](https://github.com/openedx/frontend-component-header/commit/a54f099d68b2b2771964da3326e0e7acb8383a47#diff-005d7781d3503da55f10a0f2c78af32df8774d8d94068675e88ac3426d4742cc))(Please let me know if I am missing its usage) 
- This will not affect the current usages of the frontend-component-header.

Screenshots of using custom headers for replicating different MFE headers.
Note: Screenshots are with local setup. 

Screenshot 2024-03-13 at 4 21 19 PM

Screenshot 2024-03-13 at 4 21 32 PM

Screenshot 2024-03-13 at 4 21 47 PM

Screenshot 2024-03-13 at 4 22 03 PM

Screenshot 2024-03-13 at 4 22 15 PM

Screenshot 2024-03-13 at 4 22 45 PM

### CODE REVIEW

File: docs/using_custom_header.rst [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+Props Details
+-------------
+
+The `Header` component accepts the following **optional** props for customization:
+
+``logo``
+*******
+
+The logo prop is an object containing `src`, `alt`, and `href` properties. If not passed, LOGO_URL from config will be used.
```
Comment (zainab-amir) [inline docs/using_custom_header.rst:23]: I am wondering if we can get rid of this prop? Now that I am reviewing this, I am wondering that if anyone needs to use a custom logo, they can simply override the `LOGO_URL` in their MFE config. What are your thoughts on this?

File: docs/using_custom_header.rst [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+- The ``type`` should be ``item`` or ``submenu``. If type is ``submenu``, it should contain ``submenuContent`` instead of ``href``.
+
+- If any item is to be disabled, we can pass optional ``disabled: true`` in that item object and
+
+- If any item is to be active, we can pass optional ``isActive: true`` in that item object
+
+secondaryMenuItems
+******************
+
+The secondary menu items has same structure as ``mainMenuItems``. On desktop screen, These items are displayed on the right of header just before the userMenu avatar and on mobile screen,
```
Comment (zainab-amir) [inline docs/using_custom_header.rst:69]:
```suggestion
The secondary menu items has same structure as ``mainMenuItems``. On desktop screen, these items are displayed on the right of header just before the userMenu avatar and on mobile screen,
```

**Comment by zainab-amir** [comment]
@arbrandes Can you give an estimated time on when you will be able to review this PR?

File: docs/using_custom_header.rst (hunk 1 above)
Comment (syedsajjadkazmii) [reply docs/using_custom_header.rst:23]: Thats correct. We can definitely remove it.

File: docs/using_custom_header.rst [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+=======================
+
+Overview
+--------
+
+The ``Header`` component is used to display a header with a provided ``mainMenuItems``,
+``secondaryMenuItems``, and ``userMenuItems`` props. If props are provided, the component will use them; otherwise,
+If any of the props ``(mainMenuItems, secondaryMenuItems, userMenuItems)`` are not provided, default
+items are displayed. This component provides flexibility in customization, making it suitable for a wide
+range of applications.
```
Comment (MaxFrank13) [inline docs/using_custom_header.rst:13]: Nit:

```suggestion
if any of the props ``(mainMenuItems, secondaryMenuItems, userMenuItems)`` are not provided, default
items will be displayed. This component provides flexibility in customization, making it suitable for a wide
range of applications.
```

File: docs/using_custom_header.rst [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+Props Details
+-------------
+
+The `Header` component accepts the following **optional** props for customization:
+
+``mainMenuItems``
+*****************
+
+The main menu items is a list of menu items objects. On desktop screens, these items are displayed on the left, to the right of the logo icon and to the left of the secondary menu.
```
Comment (MaxFrank13) [inline docs/using_custom_header.rst:23]:
```suggestion
The main menu items is a list of menu items objects. On desktop screens, these items are displayed on the left side next to the logo icon.
```

Original text was a little confusing to me without a visual. I don't think the secondary menu needs to be mentioned as the way this was initially written implies that there is no space in between the two menus

File: docs/using_custom_header.rst [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+       { type: 'item', href: '/help', content: 'Help' },
+   ]
+
+userMenuItems
+*************
+
+The user menu items is list of objects. On desktop screens, these items are displayed as a dropdown menu on the most right side of the header. The dropdown is opened by clicking on the avatar icon, which is typically located at the far right of the header.
+On mobile screens, the user menu is also displayed as a dropdown menu, appearing under the avatar icon.
+
+User Menu is list of objects. Each object represent a group in user menu. Each object contains the ``heading`` and
```
Comment (MaxFrank13) [inline docs/using_custom_header.rst:70]:
```suggestion
Each object represents a group in the user menu. Each object contains the ``heading`` and
```

Seems redundant given what's written on line 67

File: src/Header.jsx [hunk 6]
```diff
@@ -30,50 +31,65 @@ subscribe(APP_CONFIG_INITIALIZED, () => {
...
+ * Header component for the application.
+ * Displays a header with the provided main menu, secondary menu, and user menu when the user is authenticated.
+ * If any of the props (mainMenuItems, secondaryMenuItems, userMenuItems) are not provided, default
+ * items are displayed.
+ * For more details on how to use this component, please refer to this document:
+ * https://github.com/openedx/frontend-component-header/blob/master/docs/using_custom_header.rst
+ *
+ * @param mainMenuItems
+ * @param secondaryMenuItems
+ * @param userMenuItems
```
Comment (MaxFrank13) [inline src/Header.jsx:44]: Can we add types here? Are all of these lists/arrays?

File: docs/using_custom_header.rst (hunk 4 above)
Comment (syedsajjadkazmii) [reply docs/using_custom_header.rst:23]: makes sense

File: docs/using_custom_header.rst (hunk 5 above)
Comment (syedsajjadkazmii) [reply docs/using_custom_header.rst:70]: yes thanks.

File: docs/using_custom_header.rst [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+           submenuContent: [
+               '<div className="mb-1"><a rel="noopener" href="#">Submenu item 1</a></div>',
+               '<div className="mb-1"><a rel="noopener" href="#">Submenu item 2</a></div>'
+           ],
+       },
+   ]
+
+**Note:**
+
+- The ``type`` should be ``item`` or ``submenu``. If type is ``submenu``, it should contain ``submenuContent`` instead of ``href``.
```
[the remark this answers is not in the document]
Comment (syedsajjadkazmii) [reply docs/using_custom_header.rst:52]: yes sure. added

File: src/Header.jsx (hunk 6 above)
Comment (syedsajjadkazmii) [reply src/Header.jsx]: yes sure. added

**Comment by syedsajjadkazmii** [comment]
Hi @arbrandes, waiting for your review on this PR. :)

**Review by MaxFrank13 — APPROVED** [review]
Thanks for addressing all of the documentation nit picks! Approving from my end

File: src/DesktopHeader.jsx [hunk 8]
```diff
@@ -54,22 +60,14 @@ class DesktopHeader extends React.Component {
...
-        <MenuTrigger tag="a" className="nav-link d-inline-flex align-items-center">
-          {appMenuContent} <CaretIcon role="img" aria-hidden focusable="false" />
-        </MenuTrigger>
-        <MenuContent className="mb-0 dropdown-menu show dropdown-menu-right pin-right shadow py-2">
-          {menuItems.map(({ type, href, content }) => (
-            <a className={`dropdown-${type}`} key={`${type}-${content}`} href={href}>{content}</a>
-          ))}
-        </MenuContent>
-      </Menu>
-    );
```
Comment (arbrandes) [inline src/DesktopHeader.jsx:72]: For the record, I could not find any MFEs that use this, either.

**Pull request merged**